types/src/dkg/real_dkg/rounding/mod.rs

```diff
@@ -160,32 +182,40 @@ impl DKGRoundingProfile {
         secrecy_threshold_in_stake_ratio: U64F64,
         reconstruct_threshold_in_stake_ratio: U64F64,
         fast_secrecy_threshold_in_stake_ratio: Option<U64F64>,
-    ) -> Self {
-        assert!(total_weight_min >= validator_stakes.len());
-        assert!(total_weight_max >= total_weight_min);
-        assert!(secrecy_threshold_in_stake_ratio * U64F64::from_num(3) > U64F64::from_num(1));
-        assert!(secrecy_threshold_in_stake_ratio < reconstruct_threshold_in_stake_ratio);
-        assert!(reconstruct_threshold_in_stake_ratio * U64F64::from_num(3) <= U64F64::from_num(2));
+    ) -> anyhow::Result<Self> {
+        ensure!(total_weight_min >= validator_stakes.len());
+        ensure!(total_weight_max >= total_weight_min);
+        ensure!(secrecy_threshold_in_stake_ratio * U64F64::from_num(3) > U64F64::from_num(1));
+        ensure!(secrecy_threshold_in_stake_ratio < reconstruct_threshold_in_stake_ratio);
```

Comment:
```suggestion
        ensure!(secrecy_threshold_in_stake_ratio <= reconstruct_threshold_in_stake_ratio);
```
technically it should be `<=`? @danielxiangzl

Reply:
Yes, but in this case the total weights can be as large as total stakes in the worst case. I am fine with <=.